packages/client/src/objectSet/ObjectSetListenerWebsocket.ts

```diff
@@ -333,8 +333,12 @@ export class ObjectSetListenerWebsocket {
     if (this.#ws == null) {
       const { baseUrl, tokenProvider } = this.#client;
       const base = new URL(baseUrl);
-      const url =
-        `wss://${base.host}/api/v2/ontologySubscriptions/ontologies/${this.#client.ontologyRid}/streamSubscriptions`;
+      const url = new URL(
+        `api/v2/ontologySubscriptions/ontologies/${this.#client.ontologyRid}/streamSubscriptions`,
+        base,
+      );
+      url.protocol = url.protocol.replace("https", "wss");
```

Comment:
Can you test that this works still? I'd be fine with just an e2e or even a console log of the URL but putting this in a method and adding a test may also be nice

Reply:
TS playground looks good

Reply:
Added a helper method and test as well